lti_consumer/lti.py

```diff
@@ -0,0 +1,288 @@
+"""
+This module encapsulates code which implements the LTI specification.
+
+For more details see:
+https://www.imsglobal.org/activity/learning-tools-interoperability
+"""
+
+import logging
+import urllib
+import json
+
+from .exceptions import LtiError
+from .oauth import get_oauth_request_signature, verify_oauth_body_signature
+
+
+log = logging.getLogger(__name__)
+
+
+def parse_result_json(json_str):
+    """
+    Helper method for verifying LTI 2.0 JSON object contained in the body of the request.
+
+    The json_str must be loadable.  It can either be an dict (object) or an array whose first element is an dict,
+    in which case that first dict is considered.
+    The dict must have the "@type" key with value equal to "Result",
+    "resultScore" key with value equal to a number [0, 1],
+    The "@context" key must be present, but we don't do anything with it.  And the "comment" key may be
+    present, in which case it must be a string.
+
+    Arguments:
+        json_str (unicode):  The body of the LTI 2.0 results service request, which is a JSON string]
+
+    Returns:
+        (float, str):  (score, [optional]comment) if verification checks out
+
+    Raises:
+        LtiError (with message) if verification fails
```

Comment:
Yes. Sorry, much of this was existing code copied over from edx-platform. I did not do a thorough doc cleanup.